client/my-sites/checkout/composite-checkout/composite-checkout.tsx

```diff
@@ -598,8 +603,17 @@ export default function CompositeCheckout( {
 		feature,
 		isInEditor,
 		isComingFromUpsell,
+		isFocusedLaunch,
 	} );
 
+	const handlePaymentComplete = useCallback(
```

Comment:
🤔 I was just thinking about this and it might not do what you want in cases where the payment method is a redirect. For example, if you pay via PayPal, you're redirected away from checkout when you click to pay, then redirected directly to the thank-you page when you come back from PayPal.

Reply:
Very good point, @sirbrillig 🙏🏼 

Fixed in https://github.com/Automattic/wp-calypso/pull/48452/commits/d163f32a8ec903b8236d4a9c20cf530c31ffe587

Here is a demo: https://cloudup.com/cWO7-K6SAEH

Now I guess we need to add exceptions for eCommerce plan and registered domains where we should use the standard Checkout redirect to thank-you pages for their functionality. Do you know other cases like these?

Reply:
> Now I guess we need to add exceptions for eCommerce plan and registered domains where we should use the standard Checkout redirect to thank-you pages for their functionality. Do you know other cases like these?

Hm... not off the top of my head. You could look through [the tests that cover the thank-you url function](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js) which in theory document all the possible conditions but while I try to shepherd the thank-you functionality, the actual logic is made by other teams (largely @Automattic/martech).